### PR TITLE
docs(topics): added details on reliable event handling

### PR DESCRIPTION
**Pull request by jyecusch**

### CODE REVIEW

File: docs/messaging/topics.md [hunk 1]
```diff
@@ -65,7 +65,46 @@ loopTopic.subscribe(async (ctx) => {
 });
 ```
 
-The limitation exists to avoid infinite loops in deployed functions, where a function calls itself indirectly via a topic. These sorts of mistakes can lead to large unintentional cloud charges - something you probably want to avoid.
+The limitation exists to avoid infinite loops in deployed functions where a function calls itself indirectly via a topic. These sorts of mistakes can lead to large unintentional cloud charges - something you probably want to avoid.
+
```
Comment (raksiv) [inline docs/messaging/topics.md:69]: Can we change this language to 'The limitiation exists to protect you from infinite loops'

File: docs/messaging/topics.md [hunk 2]
```diff
@@ -65,7 +65,46 @@ loopTopic.subscribe(async (ctx) => {
...
+Basically, your publishers needs to update your database _and_ publish associated events. If a database update fails, the events should _never_ be sent. If the database update succeeds, the events should _always_ publish. The two shouldn't occur independently (i.e. one shouldn't fail while the other succeeds).
+
+One solution to this problem is the [Transactional Outbox Pattern](https://microservices.io/patterns/data/transactional-outbox.html).
+
+### Idempotent subscribers
+
+Events can be delivered more than once, but they should only be _processed_ once. To do this your subscribers need to identify and disregard duplicate events.
+
+Usually checking for duplicate payloads or IDs is enough. When you receive an event you've seen before, just don't process it, skip straight to returning a success response from your subscriber.
```
Comment (raksiv) [inline docs/messaging/topics.md:90]: remove 'just' to improve sentence flow.

**Review by raksiv — APPROVED** [review]
Added some minor non blocking suggestions.

File: docs/messaging/topics.md (hunk 1 above)
Comment (jyecusch) [reply docs/messaging/topics.md:69]: changed

File: docs/messaging/topics.md (hunk 2 above)
Comment (jyecusch) [reply docs/messaging/topics.md]: done

**Pull request merged**